### ISSUE TITLE
Feature Request: Auto-switch active project upon detected changes

### ISSUE THREAD

**Issue by purplespider** [bdkjones/CodeKit#616]
Currently, CodeKit will initiate a build whenever it detects a change in a monitored file in any one of the enabled projects. When it does this, **it would be really handy if CodeKit could also automatically make that project the active project**, if it isn't already. As this would then automatically switch the built-in server to server the project that was just updated.

This would be very useful as when switching between projects in my editor and making a change, I wouldn't have to manually go into CodeKit and change the active project, just to preview the changes.

**Comment by purplespider** [comment]
Still think this would be very useful. It feels clunky to have to switch to CodeKit, locate the project I'd like to activate for the preview server and select it. When this could happen automatically. Most of the time CodeKit is sitting out of my way in the background just doing its job, and this improvement would take this even further.

**Comment by bdkjones** [comment]
This is what the editor plug-ins do. Which editor do you use?

-Bryan

**Comment by purplespider** [comment]
Oh!? I use VS Code.

**Comment by purplespider** [comment]
Ah doesn't look like there is a plug-in for VS Code. ☹️

Perhaps adding the ability for CodeKit to auto-switch project when it detects a changed file could be used to work around some of the limitations/absence of the plugins? e.g. the limitations in Nova of not auto-switching when multiple windows are open and only being able to look for the config file in the Workspace root. (My projects' CodeKit root is usually different from the app root).

**Comment by bdkjones** [comment]
I don’t think VSCode has a plugin yet. Nova does, as do Sublime, Atom, etc.

**Comment by purplespider** [comment]
I still feel seamless auto-switching of the active project upon a detected file change would be a great addition to CodeKit. Especially as there isn't a plugin for VS Code yet (the [most popular](https://2020.stateofcss.com/en-US/other-tools/#text_editors) editor for front-end web work by far).

**Comment by bdkjones** [comment]
<img width="709" alt="Screen Shot 2021-12-17 at 01 14 30" src="https://user-images.githubusercontent.com/926566/146519885-743ed2dc-49fb-40af-b888-79a4b240843a.png">

I had to move the entire Preferences window to AutoLayout to add this one checkbox, because this Nib hasn't been touched since like Xcode 5 and Xcode 13 did not like that. Implemented for the next release.

It's off by default. I did not test what happens if you save 15 files from 15 different projects at the same time. I recommend that you also not test what happens if you save 15 files from 15 different projects at the same time.
